packages/flutter_tools/templates/plugin/lib/projectName_web.dart.tmpl

```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+// In order to *not* need this ignore, consider extracting the "web" version
+// of your plugin as a separate package, instead of inlining it in the same
+// package as the core of your plugin.
+// ignore: avoid_web_libraries_in_flutter
+import 'dart:html' as html show window;
+
+import 'package:flutter/services.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+
+/// A web implementation of the {{pluginDartClass}} plugin.
+class {{pluginDartClass}}Web {
+  static void registerWith(Registrar registrar) {
+    final MethodChannel channel = MethodChannel(
+      '{{projectName}}',
+      const StandardMethodCodec(),
+      registrar.messenger,
+    );
+
+    final pluginInstance = {{pluginDartClass}}Web();
+    channel.setMethodCallHandler(pluginInstance.handleMethodCall);
+  }
+
+  /// Handles method calls over the MethodChannel of this plugin.
+  /// Note: Check the "federated" architecture for a new way of doing this:
+  /// https://flutter.dev/go/federated-plugins
+  Future<dynamic> handleMethodCall(MethodCall call) async {
+    switch (call.method) {
+      case 'getPlatformVersion':
+        return getPlatformVersion();
+        break;
+      default:
+        throw PlatformException(
+          code: 'Unimplemented',
+          details: '{{projectName}} for web doesn\'t implement \'${call.method}\'',
+        );
+    }
+  }
```

Comment:
Can we just reuse the same method channel implementation in the <plugin_name>.dart and this web class just listen to the channel message?

Reply:
I don't think so. If I modify the <plugin_name>.dart this script becomes backwards incompatible, right?

Reply:
So we keep the <plugin_name>.dart the same. And the `web.dart` file just listens to the same method channel like other platforms do.

Reply:
Correct, in this case the web implementation listens to the same method channel. The fact that they both live right next to each other in the directory hierarchy is confusing. Imagine this `web.dart` lives in a completely separate package.

Reply:
Again, my mistake. I was looking at something else. This is good. thanks